File: ssh/src/com/anders/ssh/aop/spring/AroundAdvice.java
package com.anders.ssh.aop.spring;

import org.aopalliance.intercept.MethodInterceptor;
import org.aopalliance.intercept.MethodInvocation;

public class AroundAdvice implements MethodInterceptor {
	public Object invoke(MethodInvocation invocation) throws Throwable {
		System.out.println("开始: " + invocation.getMethod().getName());
		Object result = invocation.proceed();
		System.out.println("结束: " + invocation.getMethod().getName());
		return result;
	}
}

File: ash/ash/Aop/AroundAdvice.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Web;
using AopAlliance.Intercept;

namespace ash.Aop
{
    public class AroundAdvice : IMethodInterceptor
    {
        public object Invoke(IMethodInvocation invocation)
        {
            Console.WriteLine("开始:  " + invocation.TargetType.Name + "." + invocation.Method.Name);
            object result = invocation.Proceed();
            Console.WriteLine("结束:  " + invocation.TargetType.Name + "." + invocation.Method.Name);
            return result;
        }
    }
}
